bindings/pyroot/src/Pythonize.cxx

```diff
@@ -2262,6 +2263,19 @@ namespace {
       return BindCppObject( addr, (Cppyy::TCppType_t)Cppyy::GetScope( "TObject" ), kFALSE );
    }
 
+   //- Pretty printing with cling::PrintValue
+   PyObject *ClingPrintValue(ObjectProxy *self)
+   {
+      PyObject* cppname = PyObject_GetAttrString((PyObject*) self, "__cppname__");
+      std::string className = PyROOT_PyUnicode_AsString(cppname);
+      Py_XDECREF(cppname);
+
+      std::stringstream calcPrintValue;
+      calcPrintValue << "cling::printValue((" << className << "*)" << self->GetObject() << ");";
+      auto pprint = (std::string*) gInterpreter->Calc(calcPrintValue.str().c_str());
```

Comment:
You can't know :-) but this adds a temporary with process lifetime. You could instead do
```
calcPrintValue << "*((std::string*)" << &localString << ") = cling::printValue((" << className << "*)" << self->GetObject() << ");";
```
to assign the C++ string within the interpreter, instead of relying on the interpreter to extend the return value's lifetime.

Reply:
Done :+1: